pkg/volume/util/nestedpendingoperations/nestedpendingoperations_test.go

```diff
@@ -264,6 +265,41 @@ func Test_NewGoRoutineMap_Negative_SecondOpBeforeFirstCompletes(t *testing.T) {
 	}
 }
 
+func Test_NewGoRoutineMap_Negative_SecondThirdOpWithDifferentNames(t *testing.T) {
+	// Arrange
+	grm := NewNestedPendingOperations(true /* exponentialBackOffOnError */)
+	volumeName := v1.UniqueVolumeName("volume-name")
+	op1Name := "mount_volume"
+	operation1 := generateErrorFunc()
+	err1 := grm.Run(volumeName, "" /* operationSubName */, types.GeneratedOperations{OperationFunc: operation1, OperationName: op1Name})
+	if err1 != nil {
+		t.Fatalf("NewGoRoutine failed. Expected: <no error> Actual: <%v>", err1)
+	}
+	// Shorter than exponential backoff period, so as to trigger exponential backoff error on second
+	// operation.
+	time.Sleep(100 * time.Millisecond)
+	operation2 := generateNoopFunc()
+
+	// Act
+	err2 := grm.Run(volumeName, "" /* operationSubName */, types.GeneratedOperations{OperationFunc: operation2, OperationName: op1Name})
+
+	// Assert
+	if err2 == nil {
+		t.Fatalf("NewGoRoutine did not fail. Expected: <Failed to create operation with name \"%s\". An operation with that name already exists.> Actual: <no error>", volumeName)
+	}
+	if !exponentialbackoff.IsExponentialBackoff(err2) {
+		t.Fatalf("NewGoRoutine did not return exponential backoff, got: %v", err2)
+	}
```

Comment:
Nit: it's possible to join these two `if`'s (e.g., keep this one and use previous message).

Reply:
I am leaving this as it. I would like to know if error was nil vs if error was really exponential backoff.